### PR TITLE
Implemented `markThreadRead`

### PR DESCRIPTION
**Pull request by AlexVincent525**
Already done formatting with `dartfmt`.

Maybe use `StatusCodes.RESET_CONTENT` instead of `205` would be better?

### CODE REVIEW

File: lib/src/common/activity_service.dart [hunk 1]
```diff
@@ -180,7 +180,18 @@ class ActivityService extends Service {
       _github.getJSON("/notification/threads/$threadId",
           statusCode: StatusCodes.OK, convert: Notification.fromJSON);
 
-  // TODO: Implement markThreadRead: https://developer.github.com/v3/activity/notifications/#mark-a-thread-as-read
+  /// Mark the specified notification thread as read.
+  ///
+  /// API docs: https://developer.github.com/v3/activity/notifications/#mark-a-thread-as-read
+  Future<bool> markThreadRead(String threadId) {
```
Comment (robrbecker) [inline lib/src/common/activity_service.dart]:
```suggestion
  Future<bool> markThreadRead(int threadId) {
```

File: lib/src/common/activity_service.dart [hunk 2]
```diff
@@ -180,7 +180,18 @@ class ActivityService extends Service {
...
+    return _github
+        .request("PATCH", "/notifications/threads/$threadId")
+        .then((response) {
+      return response.statusCode == 205;
```
Comment (robrbecker) [inline lib/src/common/activity_service.dart]:
```suggestion
      return response.statusCode == 205;
```

**Comment by AlexVincent525** [comment]
@robrbecker As the codes in this repo written (`getThread`)
https://github.com/SpinlockLabs/github.dart/blob/f45596901f3fc2970ba944f3b6faa08c6208bcad/lib/src/common/activity_service.dart#L179
`threadId` is `String`, so if this was different, that might require some changes to the original one.
Another thing is there's a lot of hardcoding status code, for now it might be a good start to using the constants.

File: lib/src/common/activity_service.dart (hunk 1 above)
Comment (AlexVincent525) [reply lib/src/common/activity_service.dart]: This one needs further investigation.

**Pull request merged**